docs/aws/audit/esmonitoring/rules/es_upgrade_available_remediation.mdx

```diff
@@ -1,6 +1,211 @@
+           print(f"Update initiated for domain {domain}: {update_response}")
+   ```
+
+By following these steps, you can ensure that your ElasticSearch domains are always using the latest service software, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service dashboard. You can do this by typing "Elasticsearch Service" into the search bar at the top of the console, then selecting "Elasticsearch Service" from the dropdown menu.
+3. In the Elasticsearch Service dashboard, you will see a list of your Elasticsearch domains. Click on the name of the domain you want to check.
+4. In the domain overview page, look for the "Elasticsearch version" field. This will tell you the current version of the Elasticsearch service software that the domain is using. If it is not the latest version, then this is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the ElasticSearch domains.
+
+2. Once the AWS CLI is set up, you can list all the ElasticSearch domains using the following command:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all the ElasticSearch domains.
+
+3. To get the detailed configuration of each domain, use the following command:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name your_domain_name
+   ```
+   Replace 'your_domain_name' with the actual domain name from the list obtained in the previous step. This command will return a detailed configuration of the specified ElasticSearch domain.
+
+4. In the output of the above command, look for the 'ElasticsearchVersion' field. This field indicates the version of the ElasticSearch service software used by the domain. If the version is not the latest, it indicates a misconfiguration. You need to manually compare this version with the latest version available on the official ElasticSearch website.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing the script, you need to install the necessary Python libraries. For this task, you will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and establish a session: After installing the necessary libraries, you need to import them into your script and establish a session with AWS.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>